doc/concepts.qbk

```diff
@@ -199,6 +199,73 @@ the generator, for example to re-run a test suite at a later time.]
 
 [endsect]
 
+[section Quasi-Random Number Generator]
+
+A quasi-random number generator is a __UniformRandomNumberGenerator which
+provides a deterministic sequence of quasi-random numbers, based on some
+algorithm and internal state. [classref boost::random::niederreiter_base2
+Niederreiter Base 2] generator is an example of such a [qrng quasi-random
+number generator]. The "quasi" modifier is used to denote more clearly that the
+values produced by such a generator are neither random nor pseudo-random, but
+they form a low discrepancy sequence. The intuitive idea is that a low discrepancy 
+sequence is more evenly distributed than a pseudo random sequence would be.
+For example, if we generate a low discrepancy sequence of 2D points on a square,
+this square would be covered more evenly, and the number of points falling to any
+part of the square would be proportional to the number of points in the whole square.
+Such sequences share some properties of  random variables and in certain applications 
+such as the quasi-Monte Carlo method  their lower discrepancy is an important advantage.
+
+[note Quasi-random sequences are known to give efficient numerical integration 
+rules in many Bayesian statistical problems where the posterior distribution can be 
+transformed into periodic functions on a hypercube.]
+
+Harold Niederreiter gives an extensive overview on random number generation
+and quasi-Monte Carlo methods in his book "Random number generation and
+quasi-Monte Carlo methods, Society for Industrial and Applied Mathematics, 1992". 
+
+In addition to the __UniformRandomNumberGenerator requirements,
+a quasi-random number generator has some additional requirements. In the
+following table, `X` denotes a quasi-random number generator class, `u` is a value of `X`, 
+`v` is a const value of `X`, and `j` a value of type `unsigned long long`.
+
+[table QuasiRandomNumberGenerator requirements
+  [[expression] [return type] [pre/post-condition]]
+  [[`X(s)`] [-] [creates an s-dimensional generator with a default seed. Dimension `s` is an integer no less than 1.]]
+  [[`v.dimension()`] [std::size_t] [the dimension of quasi-random domain.]]
+  [[`u.seed(i)`] [`void`] [seeds the generator with the integer `i`.]]
+  [[`u.discard(j)`] [`void`] [Advances the generator by `j` steps as if by
+                              `j` calls to `u()`.]]
+ ]
+
+[note The `operator()` returns a successive element of an s-dimensional (`s` = `v.dimension()`) vector 
```

Comment:
Hmm, this sounds like a lot of data movement. It's might make more sense to allow `operator()` to accept a forward iteration and have the routine fill a region of memory provided by the user.

Reply:
I already implement this interface in all QRNGs
```
    template<class Iter>
    void generate(Iter first, Iter last) {/*..*/}
```
However, you might be surprised what you will get back. Gray-coded sequences will generate plain integers for you. You will need to do your own arithmetic to scale this to [0;1] range.

Reply:
If I'm not mistaken, the Faure sequence populate the container with some `RealType`, whereas the Sobol and Niederreiter will populate them with integer types. This will make generic programming difficult; for instance a quasi-Monte Carlo integrator will not be able to swap out the quasi-random sequence. Can we instead do the rescaling in `generate`, and have (say) `generate_integer` for integer types?